indexing-service/src/test/java/org/apache/druid/indexing/common/task/concurrent/ConcurrentReplaceAndAppendTest.java

```diff
@@ -725,6 +725,7 @@ public void testMultipleGranularities()
     // Append segment for Oct-Dec
     final DataSegment segmentV02 = asSegment(pendingSegment02);
     appendTask2.commitAppendSegments(segmentV02);
+    appendTask2.finishRunAndGetStatus();
```

Comment:
Why is this change needed?

Reply:
It is needed to trigger clean up of pending segments.
Having extra pending segments can lead to gaps in the ids of committed segments, causing tests to fail.